### DISCUSSION TITLE
[Flatpak]There are no icons in menus on GNU/Linux Mint 21.1 Cinnamon

### DISCUSSION THREAD

**Discussion by Coockie** [transmission/transmission#5453] [Support]
### What is the issue?

Hi again. I created issue here https://github.com/flathub/com.transmissionbt.Transmission/issues/64 for flatpak version transmission and they don't know correct behavior.

### Which application of Transmission?

GTK+ app on Linux, BSD, etc.

### Which version of Transmission?

4.0.3

**Comment by ghost** [comment]
Hey there @Coockie  this may be a theming issue specific to flatpak and not transmission, whats your icon theme and is this behaviour same in other flatpak apps too ?

**Comment by Coockie** [comment]
>whats your icon theme

I have default Mint 21.1 Theme - Mint-Y-Aqua for icons and applications and Mint-Y-Dark-Aqua for desktop. I changed themes to Mint-X and Adwaita but this doesn't help.

>is this behaviour same in other flatpak apps too ?

I don't know because i have only one flatpak application installed.

**Comment by Coockie** [comment]
I checked flatpak qbittorrent 4.5.2 and there was no problem with icons in menus.

**Comment by Coockie** [comment]
Also i checked flatpak GIMP 2.10.34 with org.gnome.Platform 43 and there was no problem with icons in menu.

**Comment by ghost** [comment]
@Coockie   Hmm interesting but this is surely related to flatpak only, can i get the output of 'flatpak list'  ??

Meanwhile few things you can try -->   https://itsfoss.com/flatpak-app-apply-theme/

--> https://github.com/refi64/stylepak  #Dont use this if you arent sure but will help you with later if needed

Read the guide and try to implement its suggestions if possible but carefully without destroying your existing configs etc.

**Comment by Coockie** [comment]
@CapeCrusader321 Here is a screenshot of flatpak -list
![Flatpak_list](https://user-images.githubusercontent.com/6870054/233631213-78d70fe5-1eff-4d04-b89b-2ca2ef246922.png)

**Comment by Coockie** [comment]
I have empty ~/.themes/ and ~/.icons/ directories so i copy Mint-Y-Aqua folders from /usr/share/themes and /usr/share/icons 
and i make  necessary actions from https://itsfoss.com/flatpak-app-apply-theme/ and this doesn't make icons in menu to appear. See screenshot
![Transmission_Mint-Y-Aqua_2](https://user-images.githubusercontent.com/6870054/233780690-bed74949-d55e-4d19-a72e-81f630da8d00.png)

**Comment by ghost** [comment]
> 

This shouldnt have been the case, the problem i understand is that somehow transmission is unable to render your icons in the flatpak sandboxing, in the meantime have you tried other formats like the snap one ??, i am researching this and will update you if i get something..
Also can i please get a screenshot of the qbitorrent with icons ??

**Comment by Coockie** [comment]
Here is a qbittorrent screenshot.
![qBittorrent_Flatpak](https://user-images.githubusercontent.com/6870054/233834236-5e55071c-eed0-4db4-b55c-6c4750b4853a.png)
I have Mint 21.1 with blocked snap feature that i don't know how to enable.

**Comment by Coockie** [comment]
I got snap to work and here are screenshots of transmission-gtk 4.0.3
![Transmission_GTK_Snap](https://user-images.githubusercontent.com/6870054/233843514-c90e3624-8737-4364-afd5-41dceb66e4fc.png)
![Transmission_GTK_Snap_2](https://user-images.githubusercontent.com/6870054/233843522-d039e451-5fd5-40cf-9801-2952b5731fa0.png)
As you can see there are some white space before icons but icons are here.

**Comment by ghost** [comment]
> I got snap to work and here are screenshots of transmission-gtk 4.0.3 ![Transmission_GTK_Snap](https://user-images.githubusercontent.com/6870054/233843514-c90e3624-8737-4364-afd5-41dceb66e4fc.png) ![Transmission_GTK_Snap_2](https://user-images.githubusercontent.com/6870054/233843522-d039e451-5fd5-40cf-9801-2952b5731fa0.png) As you can see there are some white space before icons but icons are here.

Well try changing your icon theme to something more popular like papirus ?? For snap papirus support after installing from apt or the ppa etc. do sudo snap install icon-theme-papirus, and for flatpak reply me back as it should automatically theme though i believe if the snap works fine adopt it over the flatpak one instead of messing up your setup ??

P.S. The whitespace in Snap is for tick marks etc. and is not a lack or bug....

**Comment by Coockie** [comment]
I tried change snap icon theme to papirus but it was not change at all even for Snap Store. See screenshots.
Snap Store no papirus.
![Snap_Store_NoPapirus](https://user-images.githubusercontent.com/6870054/234043164-36030fa4-506a-46d0-a8fc-f47dd88a2b64.png)
Snap Store papirus
![Snap_Store_Papirus](https://user-images.githubusercontent.com/6870054/234043340-35d5e43f-1b76-4f14-9d5c-dfeaeb9aed07.png)
Transmisssion Snap no Papirus
![Transmission_Snap_NoPapirus](https://user-images.githubusercontent.com/6870054/234043440-9ff51592-7993-4245-bc2e-09dc278371da.png)
Transmission Snap papirus
![Transmission_Snap_Papirus](https://user-images.githubusercontent.com/6870054/234043526-4a12cbb0-b197-4528-826d-d2afc06f8a93.png)

**Comment by Coockie** [comment]
>P.S. The whitespace in Snap is for tick marks etc. and is not a lack or bug....

What are marks you talking about? In Transmission 3.0 from repository there are no whitespace see screenshot.
![Transmission_30_deb](https://user-images.githubusercontent.com/6870054/234047009-c037a308-3f52-4259-8890-b70d70dd040a.png)

**Comment by ghost** [comment]
@Coockie  did you read my comment properly above, after installing papirus from apt you had to download icon-theme-papirus (sudo snap install icon-theme-papirus) too, since snap is confined it cant use your native icons so you will have to provide your native icons in the snap sandbox too.
As for the whitespace and ticks, in some places like the view tab ticks appears, but in the other cases its like default, and the one in repo is version 3 with gtk3 and we are talking of version 4 with gtk4 maybe some design changes.
Anyway here is how snap transmission looks with papirus if you follow my steps :


![sc10](https://user-images.githubusercontent.com/100503303/234283499-2b99ac59-9bdd-4351-9a16-01670baf965a.png)

**Comment by Coockie** [comment]
I installed papirus icon theme from ppa as described here https://github.com/PapirusDevelopmentTeam/papirus-icon-theme and snap icon theme papirus then after changing theme in Mint settings i've got these:

Transmission 3.0 from repository
![Transmission_30_deb_Papirus](https://user-images.githubusercontent.com/6870054/234346914-9976a301-bbfc-4324-b783-2f17d8d648d7.png)
Transmission Snap papirus(as you can see there are no "exe" torrent  icons but icons in menu are there)
![Transmission_Snap_Papirus_3](https://user-images.githubusercontent.com/6870054/234346989-0d0b48ba-8adb-4a63-97a9-697cd032d630.png)
Transmission 4.0.3 flatpak(as you can see there are no "exe" and "iso" torrent icons)
![Transmission_Flatpak_Papirus_1](https://user-images.githubusercontent.com/6870054/234347457-81a6c1e7-bb5b-4031-a618-35e9c22ffa53.png)
Transmission 4.0.3 flatpak(As you can see there are no icons in menu)
![Transmission_flatpak_Papirus_2](https://user-images.githubusercontent.com/6870054/234347818-ca0c86f3-06f1-443a-a592-83052c4db08c.png)

**Comment by ckerr** [comment]
I'm moving this to Discussion > Support since it doesn't appear to be a Transmission bug.

If there _is_ an actionable bug in Transmission that we can work on to make life better for packagers, then please do file a bug report with actionable information. 

If the problem is in packaging, as it seems, then it makes more sense to discuss this downstream at the packaging level rather than with Transmission. However if having a discussion thread here helps get to the root problem, that's great too.

**Comment by ghost** [comment]
@Coockie  i suggest you to adopt the snap ones for now ??

**Comment by Coockie** [comment]
@CapeCrusader321 The snap ones like a flatpak ones do not show me icons in torrents downloads as show here https://user-images.githubusercontent.com/6870054/234346989-0d0b48ba-8adb-4a63-97a9-697cd032d630.png

Also i tried snap transmission-qt and it doesn't show me icons torrents downloads and transmission-qt hasn't applied papirus icon theme in menu.Also as you can see there are no whitespace in menu before icons
![Transmission-qt](https://user-images.githubusercontent.com/6870054/236451412-308fe1c9-6788-4601-8e88-2020a11527a2.png)

**Reply by ghost** [reply]
> @CapeCrusader321 The snap ones like a flatpak ones do not show me icons in torrents downloads as show here https://user-images.githubusercontent.com/6870054/234346989-0d0b48ba-8adb-4a63-97a9-697cd032d630.png
> 
> Also i tried snap transmission-qt and it doesn't show me icons torrents downloads and transmission-qt hasn't applied papirus icon theme in menu.Also as you can see there are no whitespace in menu before icons ![Transmission-qt](https://user-images.githubusercontent.com/6870054/236451412-308fe1c9-6788-4601-8e88-2020a11527a2.png)

Let me breakdown, first unlike flatpak one snap one atleast shows icon in menus etc. and doesnt show icons for exe and iso files, as for the qt one well qt integration always has problems, the qt one only works fine for kde plasma most probably, it wont work well in Gtk DEs due to snap confinement, my opinion was that as compared to flatpaks snap one is a better option in regards to theming atleast plus for the whitespace i still dont have an valid answer for why it occurs but i am happy to fix if someone gives me insight regarding it.
I will also try fixing snap papirus icon theme if possible to render icons properly for the snap ones.
